databricks/provider.go

```diff
@@ -197,6 +218,41 @@ func providerConfigureAzureClient(d *schema.ResourceData, providerVersion string
 	return &dbClient, nil
 }
 
+// tryDatabricksCliConfigFile sets Host and Token from ~/.databrickscfg file if it exists
+func tryDatabricksCliConfigFile(d *schema.ResourceData, config *service.DBApiClientConfig) error {
+	configFile, err := homedir.Expand(d.Get("config_file").(string))
```

Comment:
1. I would let user provide the full path to the config file. In that case you do not need `homedir` package.
2. If `d.Get("config_file")` does not return string here the code is going to panic I believe.

Reply:
regarding 2 - terraform schemas handle the typing already, so I would assume it's safe to rely on those.